blockstore/splitstore/markset_badger.go

```diff
@@ -184,8 +186,53 @@ func (s *BadgerMarkSet) Has(c cid.Cid) (bool, error) {
 		return false, nil
 
 	default:
-		return false, xerrors.Errorf("error checking badger markset: %w", err)
+		return false, err
+	}
+}
+
+// writer holds the exclusive lock; put releases it
+func (s *BadgerMarkSet) put(key string) error {
+	s.pend[key] = struct{}{}
+	if len(s.pend) < badgerMarkSetBatchSize {
+		s.mx.Unlock()
```

Comment:
:bomb:

Reply:
Folding this function back into the caller is an acceptable solution. Although I'd consider breaking it up into the part that needs the lock, and the part that doesn't.

Reply:
we have to break it and scope the lock externally; the whole reason for releasing the lock in here is to avoid doing the actual write while holding the lock.

Reply:
Not holding the lock while writing is good. Releasing a lock across a function boundary is not good.

Reply:
understood, allergic reaction :)
I am changing it to release the lock locally and use a separate write function.

Reply:
done.